### ISSUE TITLE
CREATE COLLATION statement produces warning

### ISSUE THREAD

**Issue by chrishajas** [greenplum-db/gpdb#5071]
### Greenplum version or build
 PostgreSQL 9.1beta2 (Greenplum Database 6.0.0-alpha.0+dev.7210.g21ed197153 build commit:21ed197153d3526eb0e9aa5ca8435c580b01
a800-oss) on x86_64-pc-linux-gnu, compiled by gcc (GCC) 6.2.0, 64-bit compiled on May 30 2018 19:51:16 (with assert checking)
(1 row)

### OS version and uname -a
centos6

Linux mdw 2.6.32-696.30.1.el6.x86_64 #1 SMP Tue May 22 03:28:18 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

### Expected behavior
Create a collation without a warning

### Actual behavior
Creates a collation with a warning. Additionally, I am unable to comment on the collation. There is a test file (https://github.com/greenplum-db/gpdb/blob/25a90396cd1c52d252a37986e12b63e8e037aa83/src/test/regress/sql/collate.linux.utf8.sql), but it does not appear to be running in CI, else it would likely fail.

### Step to reproduce the behavior
```
test=# CREATE COLLATION some_coll FROM "de_DE";
WARNING:  OID assigned for tuple in unrecognized system catalog "pg_collation"
CREATE COLLATION
test=#
```

```
test=# comment on collation some_coll is 'coll comment';
ERROR:  collation "some_coll" for encoding "UTF8" does not exist  (seg0 192.168.5.246:20000 pid=9841)
```

**Comment by tvesely** [comment]
This issue will be resolved with #5016. The OID assignment warning is due to the `oid_dispatch` code not being trained to understand `pg_collation`.

**Comment by tvesely** [comment]
#5016 has been merged, so this issue has been resolved.
